src/lib.rs

```diff
@@ -146,6 +148,28 @@ impl ToolFamily {
             ToolFamily::Clang => "-E",
         }
     }
+
+    /// What the flags to enable all warnings
+    fn warnings_flags(&self) -> &'static [&'static str] {
+        static MSVC_FLAGS: &'static [&'static str] = &["/Wall"];
+        static GNU_FLAGS: &'static [&'static str] = &["-Wall", "-Wpedantic", "-Wextra"];
+        static CLANG_FLAGS: &'static [&'static str] = &["-Weverything"];
+
+        match *self {
+            ToolFamily::Msvc => &MSVC_FLAGS,
+            ToolFamily::Gnu => &GNU_FLAGS,
+            ToolFamily::Clang => &CLANG_FLAGS,
+        }
```

Comment:
I am not sure using `-Wpedantic` is a good idea... It rejects (an error not a warning) some programs using [non standard features](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html). Since this is by default it will break many crates using `gcc-rs` since it is not common for C code to strictly follow ISO C standard.

Also, using `-Weverything` with clang is not necessarily a good idea either. There is some warnings that do not make sense in some programs. It will just spam the output when compiling for no good reason and it will probably get people used to see warnings, thus they will ignore them...